### ISSUE TITLE
Config file globs vs CLI glob

### ISSUE THREAD

**Issue by nschonni** [vale-cli/vale#84]
Think this might just be a docs issue, but the config file setting like from the README
```ini
# Only Markdown and .txt files; change to whatever you're using.
[*.{md,txt}]
# List of styles to load.
BasedOnStyles = proselint, write-good, Joblint
# Style.Rule = {YES, NO, suggestion, warning, error} to
# enable/disable a rule or change its level.
write-good.E-Prime = NO
```
will test all files rather than just MD/TXT. I think the docs for config https://errata.ai/vale/config/ hints that this isn't the same as `vale --glob=*.{md,txt} .` but maybe you can confirm

**Comment by jdkato** [comment]
The comment should be accurate for all versions of Vale: a config file section with `[*.{md,txt}]` as a header will apply only to `.txt` and `.md` files.

- In versions prior to v1.0.2, the config file wasn't consulted prior to *opening* files (which could cause issues like #79 since, upon opening a file, Vale ensures any required dependencies are available) but irrelevant files weren't actually *searched* for issues.
- In v1.0.2 and later, Vale doesn't even open files that don't have rules defined for them.

In either case, `--glob` takes precedence. If you define `[*.{md,txt}]` in the config file and pass `--glob='*.md'`, only Markdown files will be considered.

**Comment by nschonni** [comment]
https://github.com/MicrosoftDocs/azure-docs is my torture repo for Vale :wink: 
I'll make sure i'm actually running the latest, but I thought the last time I tried `vale .` with a config only, i had to kill it after 40 minutes and 80GB of RAM consumed on my old machine. `vale --glob=*.md .` finished correctly after a few minutes

**Comment by jdkato** [comment]
This wouldn't necessarily be unexpected with versions prior to v1.0.2, as the extra work can easily add up when dealing with large repositories. In newer versions, config-only performance should be much closer to using `--glob`.

I'd also recommend specifying a directory whenever possible (i.e., avoiding `.`), as that can significantly improve performance as well.

EDIT: Also, FWIW, passing `--ignore-syntax` will give the best possible performance -- but only `text`-scoped rules will work.

**Comment by nschonni** [comment]
Here is a better example:
```ini
# This goes in a file named either `.vale` or `_vale`.

# StylesPath = path/to/some/directory
MinAlertLevel = error # suggestion, warning or error
[*.{md,html,cs,vb,yaml,yml}] # Only Markdown and .txt files
# List of styles to load
BasedOnStyles = vale
# Style.Rule = {YES, NO, suggestion, warning, error} to
# enable/disable a rule or change its level.
vale.Editorializing = NO
vale.Redundancy = NO
vale.Uncomparables = NO
```
Run1: `time vale --glob='*.{md,html,cs,vb,yaml,yml}' .`
real    0m50.625s
user    4m29.591s
sys     0m6.685s

Run2: `time vale .`
real    54m53.736s
user    9m49.690s
sys     16m38.626s

**Comment by jdkato** [comment]
Here are my results:

```shell
# See how many total files we're dealing with:
$ find . -type f | wc -l
70987
# From the root of `azure-docs`:
$ time vale --glob='*.{md,html,cs,vb,yaml,yml}' .
43.018 total
$ time vale .
3:57.74 total
```
It appears the large size of the repository (~70,987 files) has exposed an inefficiency in Vale's handling of its config file (it's not actually checking unspecified files, though, which can be tested by creating a `.rst` file with an error).

The inefficiency was actually introduced deliberately in https://github.com/errata-ai/vale/commit/20d143190362d6cb77bd33bcc59d5812d1b73d6b as a fix for crashes related to storing compiled [`globs`](https://github.com/gobwas/glob). I imagine this wasn't caught at the time because the difference is negligible on many (most?) repositories.

I'll look into improving this (although, given the size the repo, I'd say Vale is still performing pretty well).

**Comment by jdkato** [comment]
This should be much improved in the next release; here are my current benchmarks:

```shell
$ time vale --glob='*.{md,html,cs,vb,yaml,yml}' .
43.858 total
$ time vale .
43.523 total
```
Thanks for the report.

**Comment by nschonni** [comment]
No problem. That repo is why to big, but I like it for testing docs tools for that reason 😜

**Comment by jdkato** [comment]
This should be fixed with v1.0.5.

**Comment by nschonni** [comment]
Great! Looks alot better with `time vale .`
real    0m53.385s
user    4m27.210s
sys     0m7.824s
Seems like the fan kicks in more without the `--globs`, but that might just be in my head
